scm/driver/github/webhook.go

```diff
@@ -182,6 +185,74 @@ func (s *webhookService) parsePullRequestHook(data []byte) (scm.Webhook, error)
 	return dst, nil
 }
 
+func (s *webhookService) parsePipelineHook(data []byte) (scm.Webhook, error) {
+	src := new(pipelineHook)
+	err := json.Unmarshal(data, src)
+	if err != nil {
+		return nil, err
+	}
+	dst, err := convertPipelineHook(src), nil
+	return dst, nil
+}
+
+func convertPipelineHook(src *pipelineHook) *scm.PipelineHook {
+	pr := scm.PullRequest{}
+	if len(src.WorkflowRun.PullRequests) > 0 {
+		pr = scm.PullRequest{
+			Number:  src.WorkflowRun.PullRequests[0].Number,
+			Sha:     src.WorkflowRun.PullRequests[0].Head.SHA,
+			Ref:     src.WorkflowRun.PullRequests[0].Head.Ref,
+			Source:  src.WorkflowRun.PullRequests[0].Head.Ref,
+			Target:  src.WorkflowRun.PullRequests[0].Base.Ref,
+			Link:    src.WorkflowRun.PullRequests[0].URL,
+			Created: src.WorkflowRun.CreatedAt,
+		}
+	}
+
+	return &scm.PipelineHook{
+		Repo: scm.Repository{
+			ID:        strconv.Itoa(int(src.WorkflowRun.Repository.ID)),
+			Namespace: src.WorkflowRun.Repository.Owner.Login,
+			Name:      src.WorkflowRun.Repository.Name,
+			Clone:     src.Repository.CloneURL,
+			CloneSSH:  src.Repository.SSHURL,
+			Link:      src.Repository.HTMLURL,
+			Branch:    src.Repository.DefaultBranch,
+			Private:   src.Repository.Private,
+		},
+		Commit: scm.Commit{
+			Sha:     src.WorkflowRun.HeadCommit.ID,
+			Message: src.WorkflowRun.HeadCommit.Message,
+			Author: scm.Signature{
+				Login:  src.WorkflowRun.Actor.Login,
+				Name:   src.WorkflowRun.HeadCommit.Author.Name,
+				Email:  src.WorkflowRun.HeadCommit.Author.Email,
+				Avatar: src.WorkflowRun.Actor.AvatarURL,
+			},
+			Committer: scm.Signature{
+				Login:  src.WorkflowRun.Actor.Login,
+				Name:   src.WorkflowRun.HeadCommit.Author.Name,
+				Email:  src.WorkflowRun.HeadCommit.Author.Email,
+				Avatar: src.WorkflowRun.Actor.AvatarURL,
+			},
```

Comment:
commit author and actor may not be the same person that triggered the pipeline (e.g. manual pipeline execution would have a different actor)

Reply:
updated in latest changes